aQute.libg/src/aQute/lib/filter/Filter.java

```diff
@@ -4,17 +4,66 @@
  */
 package aQute.lib.filter;
 
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.invoke.MethodType;
 import java.lang.reflect.Array;
 import java.lang.reflect.Constructor;
 import java.math.BigDecimal;
 import java.math.BigInteger;
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Collections;
 import java.util.Dictionary;
 import java.util.Hashtable;
+import java.util.LinkedHashMap;
+import java.util.List;
 import java.util.Map;
+import java.util.Map.Entry;
+
+import aQute.lib.exceptions.Exceptions;
 
 public class Filter {
-	final char			WILDCARD	= 65535;
+
+	// eliminate reflection on bnd/osgi Version during resolver operations
+	static Class<?>		BND_VERSION;
+	static MethodHandle	BND_parseVersion_MH;
+	static Class<?>		OSGI_VERSION;
+	static MethodHandle	OSGI_valuOf_MH;
+	static {
+		MethodHandles.Lookup publicLookup = MethodHandles.publicLookup();
+		try {
+			BND_VERSION = Class.forName("aQute.bnd.version.Version");
+			MethodType mt = MethodType.methodType(BND_VERSION, String.class);
+			BND_parseVersion_MH = publicLookup.findStatic(BND_VERSION, "parseVersion", mt);
+		} catch (ClassNotFoundException | IllegalAccessException | NoSuchMethodException e) {
+			BND_VERSION = null;
+			BND_parseVersion_MH = null;
+		}
+		try {
+			OSGI_VERSION = Class.forName("org.osgi.framework.Version");
+			MethodType mt = MethodType.methodType(OSGI_VERSION, String.class);
+			OSGI_valuOf_MH = publicLookup.findStatic(OSGI_VERSION, "valueOf", mt);
```

Comment:
You might consider being able to fallback to the Version(String) constructor since Version.valueOf was only added in Core R6. But that is not necessary since it will always fallback to the `else` class which will use the String constructor.

Reply:
Also, you miss the `e` in `OSGI_valuOf_MH`.

Reply:
pushed..